huaweicloud/services/ims/resource_huaweicloud_images_image_share.go

```diff
@@ -0,0 +1,280 @@
+	return &schema.Resource{
+		CreateContext: resourceImsImageShareCreate,
+		UpdateContext: resourceImsImageShareUpdate,
+		ReadContext:   resourceImsImageShareRead,
+		DeleteContext: resourceImsImageShareDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"source_image_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the ID of the source image.`,
+			},
+			"target_project_ids": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: `Specifies the ID set of the target project.`,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceImsImageShareCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	projectIds := d.Get("target_project_ids")
+	err := shareImages(ctx, d, cfg, region, projectIds.(*schema.Set).List())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(resourceId)
+
+	return resourceImsImageShareRead(ctx, d, meta)
+}
+
+func resourceImsImageShareUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	if d.HasChange("target_project_ids") {
+		oProjectIdsRaw, nProjectIdsRaw := d.GetChange("target_project_ids")
+		shareProjectIds := nProjectIdsRaw.(*schema.Set).Difference(oProjectIdsRaw.(*schema.Set))
+		unShareProjectIds := oProjectIdsRaw.(*schema.Set).Difference(nProjectIdsRaw.(*schema.Set))
+		if shareProjectIds.Len() > 0 {
+			err := shareImages(ctx, d, cfg, region, shareProjectIds.List())
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		}
+		if unShareProjectIds.Len() > 0 {
+			err := unShareImages(ctx, d, cfg, region, unShareProjectIds.List())
+			if err != nil {
+				return diag.FromErr(err)
+			}
+		}
+	}
+
+	return resourceImsImageShareRead(ctx, d, meta)
+}
+
+func resourceImsImageShareRead(_ context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
+	return nil
+}
+
+func resourceImsImageShareDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	projectIds := d.Get("target_project_ids")
+	err := unShareImages(ctx, d, cfg, region, projectIds.(*schema.Set).List())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func shareImages(ctx context.Context, d *schema.ResourceData, cfg *config.Config, region string, projectIds []interface{}) error {
```

Comment:
you don't need to input region, you can get it with cfg.GetRegion(d)

Reply:
done